p2p/protocol/identify/id.go

```diff
@@ -347,6 +347,10 @@ func (ids *idService) sendPushes(ctx context.Context) {
 			defer func() { <-sem }()
 			ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
 			defer cancel()
+
+			// We only want to send an identify push if we already have an open
+			// connection.
+			ctx = network.WithNoDial(ctx, "id push")
 			str, err := ids.Host.NewStream(ctx, c.RemotePeer(), IDPush)
```

Comment:
Should we create the stream explicitly on the connection and do multistream select explicitly like we do for identify(non push). This might use a different connection to open the stream.

Reply:
Yeah, that's better. done on b53c174b85a9